boards/nxp/ls1046ardb/board.yml

```diff
@@ -0,0 +1,10 @@
+board:
+  name: nxp_ls1046ardb
+  vendor: nxp
+  socs:
+  - name: ls1046a
+    variants:
+    - name: smp
+      variants:
+      - name: 2cores
+      - name: 4cores
```

Comment:
no, smp would be one of these e.g. building for `nxp_ls1046ardb/ls1046a/smp` would give you the 2 core version and building for `nxp_ls1046ardb/ls1046a/smp/4cores` would give you the 4 core version, so the `_smp_2cores` filenames would change to `_smp`, and only differences for the 4core ones would remain in the `_smp_4cores` files, which assumingly (I have not looked at the files) would only be the defconfig

Reply:
And the way the hwmv2 configuration works is the base configuration will be applied first, that is `nxp_ls1046ardb_defconfig`, then `nxp_ls1046ardb_ls1046a_defconfig` will be applied (does not exist so can be ignored) then `nxp_ls1046ardb_ls1046a_smp_defconfig` will be applied, then finally `nxp_ls1046ardb_ls1046a_smp_4cores_defconfig`, so anything in the base files does not need to be set again in the files overlayed on top (note: defconfig files only work like this, only one .dts file is used)

Reply:
See https://github.com/zephyrproject-rtos/zephyr/tree/collab-hwm/boards/synopsys/nsim for an example

Reply:
@nordicjm , thanks for your detail explanation, have updated, please review again.